tests/test_instance_incremental.py

```diff
@@ -23,15 +25,79 @@ def numpy_data():
 
     return (x_train, y_train.astype(int)), (x_test, y_test.astype(int))
 
-# yapf: disable
-
 
-def test_constant_class_number(numpy_data):
+@pytest.mark.parametrize("nb_tasks,nb_tasks_gt", [
+    (2, 2),
+    (6, 6),
+    (1, 1),
+])
+def test_instance_auto_nb_tasks(numpy_data, nb_tasks, nb_tasks_gt):
     train, test = numpy_data
-    dummy = InMemoryDatasetTest(*train, *test)
-    clloader = InstanceIncremental(dummy, nb_tasks=2)
+    dummy = InMemoryDataset(*train, *test)
+    clloader = InstanceIncremental(dummy, nb_tasks=nb_tasks)
+
+    nb_classes = clloader.nb_classes
+
+    assert len(clloader) == nb_tasks_gt
+    for task_id, train_dataset in enumerate(clloader):
+        assert nb_classes == len(np.unique(train_dataset.y))
+
+
+@pytest.fixture
+def numpy_data_per_task():
+    nb_classes = 6
+    nb_tasks = 3
+    nb_data = 100
+
+    x_train = []
+    y_train = []
+    t_train = []
+    for i in range(nb_tasks):
+        for j in range(nb_classes):
+            x_train.append(np.ones((nb_data, 4, 4, 3), dtype=np.uint8) * i)
+            y_train.append(np.ones(nb_data) * j)
+            t_train.append(np.ones(nb_data) * i)
+    x_train = np.concatenate(x_train)
+    y_train = np.concatenate(y_train)
+    t_train = np.concatenate(t_train)
+
+    x_test = np.copy(x_train)
+    y_test = np.copy(y_train)
+    t_test = np.copy(t_train)
+
+    return (x_train, y_train.astype(int), t_train), (x_test, y_test.astype(int), t_test)
+
+
+@pytest.mark.parametrize("nb_tasks,nb_tasks_gt,catch", [
+    (2, 2, False),
+    (6, None, True),
+    (0, 3, False),
+])
+def test_instance_default_nb_tasks(numpy_data_per_task, nb_tasks, nb_tasks_gt, catch):
```

Comment:
We should maybe add a small comment at the beginning of a test to explain what it does